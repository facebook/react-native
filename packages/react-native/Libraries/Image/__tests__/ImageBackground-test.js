/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../jest/renderer');
const ImageBackground = require('../ImageBackground');
const React = require('react');

describe('ImageBackground', () => {
  it('should render as <ImageBackground> when mocked', async () => {
    const instance = await render.create(
      <ImageBackground
        style={{width: 150, height: 50}}
        source={{uri: 'foo-bar.jpg'}}
      />,
    );
    expect(instance).toMatchSnapshot();
  });

  it('should render as <RCTImageView> when not mocked', async () => {
    jest.dontMock('../ImageBackground');

    const instance = await render.create(
      <ImageBackground
        style={{width: 150, height: 50}}
        source={{uri: 'foo-bar.jpg'}}
      />,
    );
    expect(instance).toMatchSnapshot();
  });

  it('should be set importantForAccessibility in <View> and <Image>', async () => {
    const instance = await render.create(
      <ImageBackground
        importantForAccessibility={'no'}
        style={{width: 150, height: 50}}
        source={{uri: 'foo-bar.jpg'}}
      />,
    );
    expect(instance).toMatchSnapshot();
  });
});
