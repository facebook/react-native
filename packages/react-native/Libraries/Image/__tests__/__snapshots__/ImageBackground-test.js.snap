// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBackground should be set importantForAccessibility in <View> and <Image> 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  importantForAccessibility="no"
  style={
    Object {
      "height": 50,
      "width": 150,
    }
  }
>
  <Image
    importantForAccessibility="no"
    source={
      Object {
        "uri": "foo-bar.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 50,
          "width": 150,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`ImageBackground should render as <ImageBackground> when mocked 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": 50,
      "width": 150,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "foo-bar.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 50,
          "width": 150,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`ImageBackground should render as <RCTImageView> when not mocked 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": 50,
      "width": 150,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "foo-bar.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 50,
          "width": 150,
        },
        undefined,
      ]
    }
  />
</View>
`;
